app/src/main/java/org/fossasia/openevent/app/qrscan/BarcodeTrackerFactory.java

```diff
@@ -0,0 +1,26 @@
+package org.fossasia.openevent.app.qrscan;
+
+import com.google.android.gms.vision.MultiProcessor;
+import com.google.android.gms.vision.Tracker;
+import com.google.android.gms.vision.barcode.Barcode;
+
+import org.fossasia.openevent.app.qrscan.ui.GraphicOverlay;
+
+/**
+ * Factory for creating a tracker and associated graphic to be associated with a new barcode.  The
+ * multi-processor uses this factory to create barcode trackers as needed -- one for each barcode.
+ */
+class BarcodeTrackerFactory implements MultiProcessor.Factory<Barcode> {
+    private GraphicOverlay<BarcodeGraphic> mGraphicOverlay;
+
+    BarcodeTrackerFactory(GraphicOverlay<BarcodeGraphic> barcodeGraphicOverlay) {
+        mGraphicOverlay = barcodeGraphicOverlay;
+    }
+
+    @Override
+    public Tracker<Barcode> create(Barcode barcode) {
+        BarcodeGraphic graphic = new BarcodeGraphic(mGraphicOverlay);
+        return new BarcodeGraphicTracker(mGraphicOverlay, graphic);
+    }
```

Comment:
Stuff like these were the reason dagger was made. To remove the need of factories